RecoCTPPS/PixelLocal/src/RPixPlaneCombinatoryTracking.cc

```diff
@@ -504,6 +507,124 @@ std::vector<RPixPlaneCombinatoryTracking::PointAndReferencePair >
   return sortedVector;
 
 }
+//------------------------------------------------------------------------------------------------//
+
+// The method sets the tracks recoInfo_ value for localTrackVector_ tracks
+void RPixPlaneCombinatoryTracking::addRecoInfo(int run)
+{
+  // Hardcoded shift periods: 
+  // 0 -> Before starting bx shifts.
+  // 1 -> Starting from run 300802: Sec45 St2 Rp3 Pl 0,2,3 ROC 0 shifted.
+  // 2 -> Starting from run 303338: No shift.
+  // 3 -> Starting from run 305169: Sec45 St2 Rp3 Pl 1,3,5 ROC 0 shifted.
+  // 4 -> Starting from run 305965: Sec45 St2 Rp3 Pl 1,3,5 ROC 0 shifted & Sec56 St2 Rp3 Pl 2,4,5 ROC 5 shifted.
+    
+  unsigned short shiftPeriod = 0;
+
+  // These variables hold the information of the runs when the detector was taking data in 3+3 Mode and which planes were bx-shifted
+  // These values will never be changed and the 3+3 Mode will never be used again in the future
+  const std::vector<int> periodLimits = {300802,303338,305169,305965,307083};
+  const std::map< unsigned short, std::map< CTPPSPixelDetId,std::vector<bool> > > isPlaneShifted =
```

Comment:
Right, strictly speaking these are conditions data. We propose this straight-forward solution because of their small size and the fact that they will never change (this scheme proved not useful and will not be repeated).

Reply:
It looks like a map cannot be defined as constexpr, given the fact that std::map is not a literal type... Am I missing something?

I'll add here some information about what plane bx-shifting is below:
the tracks belonging to runs in the periods listed in the map have been collected with an alternative readout mode of the detector. 
Since we know that the pixels readout electronics start responding with a delay when they get irradiated, it was tried to set the most irradiated ROC of 3 planes (the ones marked with the "true" value) to acquire data belonging to bx+1, while the other 3 acquire on bx. This was ideally done to "catch" signals that cross the threshold late with respect to bx. 
It has nothing to do with detector positioning, it only affects the detector timing settings.
This action was taken only for this limited number of runs in pixel detectors, therefore there is no need to compute the recoInfo parameter also for other tracks. It's only purpose is to provide an information about which planes (shifted or not) contributed to the track.
If more detailed information is needed I will surely provide it

Reply:
> It looks like a map cannot be defined as constexpr, given the fact that std::map is not a literal type... Am I missing something?

static const should work then

> 
> I'll add here some information about what plane bx-shifting is below:
...
so, to rephrase, some of the detectors may have a significantly different response and this is the information useful for analysis. OK.
Thanks for clarifying the details.